src/core/thread/mle_router.cpp

```diff
@@ -4046,6 +4050,8 @@ void MleRouter::HandleAddressSolicitResponse(Coap::Message *         aMessage,
 
     SuccessOrExit(Tlv::ReadUint8Tlv(*aMessage, ThreadTlv::kStatus, status));
 
+    mAddressSolicitRejected = (status != ThreadStatusTlv::kSuccess);
```

Comment:
move it into below `if` which does status check?

Reply:
This also sets `mAddressSolicitRejected` to `TRUE` when status == `kSuccess`. 
Even though I think it might  not be necessary.

Reply:
you should mean `mAddressSolicitRejected` is `FALSE` when status is `kSuccess`, which should be done already when the device becomes child, that is, before address solicit request 

https://github.com/openthread/openthread/blob/4cb5d1b30acbdebebe8e7ee9543afe2168c1494d/src/core/thread/mle_router.cpp#L232-L237

Given it is not necessary to clear here, prefer to put a simple sentence in below `if`

Reply:
Revised accordingly. 

I meant that when `status == kSuccess`, the original code sets `mAddressSolicitRejected` to `FALSE`, which reflects the fact that it is not rejected by the Leader. 
However, after the REED upgrade to Router, the `mAddressSolicitRejected` flag is not used anywhere. So, whether or not setting it to `FALSE` on `status=kSuccess` has no impact.